src/boardState.ts

```diff
@@ -0,0 +1,239 @@
+import {ShipPlacement} from './shipPlacement';
+import {ShipPossibilities} from './shipPossibilities';
+import * as Globals from './constants';
+import {Position} from './position';
+import {Shot} from './shot';
+import {randBetween} from './random'
+import {ShotImpliedPlacements} from './shotImpliedPlacements'
+
+export class BoardState{
+    public ships: Array<ShipPlacement> = [];
+    public shots: Array<Shot> = [];
+
+    public constructor(ships: Array<ShipPlacement>, shots: Array<Shot> = []){
+        this.ships = ships;
+        this.shots = shots;
+    }
+
+    static tryGetRandomSet(shipLengths: Array<number>, shots:Array<Shot> = [], hitShots: Array<Shot> = [], pathToFollow: number[] = [], exhaustive:boolean = false):{state: BoardState, found: boolean, nextPath: number[]}
```

Comment:
This is a very long method. I think it could almost be extracted into it's own class, with the variables set up just below as class vars via the constructor, and a bunch of methods such as `getValidShipPositions`, `getHits`, etc..